### PR TITLE
Add support for negative numbers in Aim query syntax

### LINKED ISSUE #416: Negative numbers are not supported in Aim query syntax
The query syntax in Aim UI does not accept negative numbers: `run.metrics['my_metric'].last < -1` gives `Error:Query syntax error at line (1, 35)`

### PR DESCRIPTION
**Pull request by jgiannuzzi**
Fixes #416.

Also reworked the query unit tests to be more readable and more focused on the actual query parsing.

### CODE REVIEW

File: pkg/api/aim/query/query_test.go [hunk 1]
```diff
@@ -42,47 +42,95 @@ func (s *QueryTestSuite) TestPostgresDialector_Ok() {
 		expectedVars []interface{}
 	}{
 		{
-			name:         "TestRunNameWithoutFunction",
-			query:        `(run.name == 'run')`,
-			expectedSQL:  `SELECT * FROM "runs" WHERE ("runs"."name" = $1 AND "runs"."lifecycle_stage" <> $2) ORDER BY "runs"."run_uuid" LIMIT 1`,
+			name:  "TestRunNameWithoutFunction",
+			query: `(run.name == 'run')`,
+			expectedSQL: `SELECT "run_uuid" FROM "runs" ` +
```
Comment (dsuhinin) [inline pkg/api/aim/query/query_test.go:47]: Jonathan it is hard to maintain all this `+` Why do we need them? We already use `backticks`.

File: pkg/api/aim/query/query.go [hunk 2]
```diff
@@ -733,13 +733,23 @@ func (pq *parsedQuery) parseUnaryOp(node *ast.UnaryOp) (any, error) {
...
-	}
 	switch node.Op {
+	case ast.USub:
+		switch e := e.(type) {
+		case int:
+			return -e, nil
+		case float64:
+			return -e, nil
+		default:
+			return nil, fmt.Errorf("unsupported unary operation %q on %T", node.Op, e)
```
Comment (dsuhinin) [inline pkg/api/aim/query/query.go:744]: maybe `unsupported` type?

**Review by dsuhinin — COMMENTED** [review]
few comments.

File: pkg/api/aim/query/query_test.go (hunk 1 above)
Comment (jgiannuzzi) [reply pkg/api/aim/query/query_test.go:47]: the idea was to make this more readable, and to avoid lines that are too long

Comment (dsuhinin) [reply pkg/api/aim/query/query_test.go:47]: correct. but `backticks` allows you to use new lines.

File: pkg/api/aim/query/query.go (hunk 2 above)
Comment (jgiannuzzi) [reply pkg/api/aim/query/query.go]: good point!
addressed in 5005490d6375e9d6856325dfb12d7b7cd29cd866

File: pkg/api/aim/query/query_test.go (hunk 1 above)
Comment (jgiannuzzi) [reply pkg/api/aim/query/query_test.go:47]: yes but we don't want new lines to be part of the string!

**Review by dsuhinin — APPROVED** [review]

**Pull request merged**